R/importM.COSMIC.R

```diff
@@ -0,0 +1,258 @@
+                           writeToFile = FALSE,
+                           silent = FALSE,
+                           writeLog = TRUE) {
+
+
+    # ==== PARAMETER TYPE VALIDATION ==========================================
+
+    # General parameter checks
+    cR <- character()
+    cR <- c(cR, .checkArgs(fNameCNA,     like = "FILE_E",   checkSize = TRUE))
+    cR <- c(cR, .checkArgs(writeToFile,  like = logical(1), checkSize = TRUE))
+    cR <- c(cR, .checkArgs(silent,       like = logical(1), checkSize = TRUE))
+    cR <- c(cR, .checkArgs(writeLog,     like = logical(1), checkSize = TRUE))
+
+    if(length(cR) > 0) {
+        stop(cR)
+    }
+
+    if (writeToFile) {
+        # Make sure output file name is provided if writeToFile is TRUE.
+        if (missing(outFName)) {
+            stop("No output file specified in outFName.")
+        }
+
+        # Check that outFName is a string.
+        cR <- character()
+        cR <- c(cR, .checkArgs(outFName,
+                               like = "a",
+                               checkSize = TRUE))
+
+        if(length(cR) > 0) {
+            stop(cR)
+        }
+
+    } else {
+        if (!missing(outFName)) {
+            stop("outFName provided, but writeToFile is FALSE.")
+        }
+    }
+
+    # Check that file provided at path fNameCNA exists
+    if (!file.exists(fNameCNA)) {
+        stop("Parameter error:\n   File specified at fNameCNA not found.")
+    }
+
+
+    # ==== READ COSMIC CNA DATA ===============================================
+
+    if (!silent) {
+        cat("Reading CNA data from ", fNameCNA, " ...\n")
+    }
+
+    # Read header line from fNameCNA file
+    tmpHeader <- readLines(fNameCNA, n = 1)
+
+    # Create character vector of col names from header string
+    header <- unlist(strsplit(tmpHeader[1], "\t"))
+
+    # The required columns from Cosmic CNA table
+    required <- c("gene_name", "ID_SAMPLE", "ID_TUMOUR", "TOTAL_CN")
+
+    # Find the column numbers in the header that match headings in required
+    iCol <- which(header %in% required)
+
+    # Create a column mask for readr::read_delim()
+    readMask <- paste(paste(rep("_", length(header)),
+                            collapse = ""),
+                      sep = "")
+
+    # Set target columns to characters
+    for (col in iCol) {
+        substr(readMask, col, col) <- "c"
+    }
+
+    # Read only the selected columns from Cosmic CNA file
+    dataCNA <- readr::read_delim(file = fNameCNA,
+                                 delim = "\t",
+                                 col_types = readMask,
+                                 n_max = Inf)
+
+    if (!silent) {
+        cat("Done reading from ", fNameCNA, ".\n")
+    }
+
+    if (nrow(dataCNA) == 0) {
+        stop("Input error: No CNAs found in file.")
+    }
+
+    # Convert all TOTAL_CN to numeric
+    dataCNA$TOTAL_CN <- as.numeric(dataCNA$TOTAL_CN)
+
+    # Subtract 2 from each CNA count
+    if (!any(dataCNA$TOTAL_CN < 0)) {
```

Comment:
I think this condition would *seriously* violate an assumption you are making about the data. I don't think this can just pass silently. Or is it expected that they can either be positive or negative? If this is unexpected, I would test this right after import and ```stop()```. If this is expected, you need to write a note whether you subtracted -2 or not to console and to the logfile.